servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/H2PriorKnowledgeFeatureParityTest.java

```diff
@@ -322,6 +325,67 @@ void cookiesRoundTrip(HttpTestExecutionStrategy strategy, boolean h2PriorKnowled
         }
     }
 
+    @ParameterizedTest(name = "{displayName} [{index}] client={0}, h2PriorKnowledge={1}")
+    @MethodSource("clientExecutors")
+    void teHeaderOnlyAllowsTrailers(HttpTestExecutionStrategy strategy, boolean h2PriorKnowledge) throws Exception {
+        setUp(strategy, h2PriorKnowledge);
+        InetSocketAddress serverAddress = bindHttpEchoServer();
+        try (BlockingHttpClient client = forSingleAddress(HostAndPort.of(serverAddress))
+                .protocols(h2PriorKnowledge ? h2Default() : h1Default())
+                .executionStrategy(clientExecutionStrategy).buildBlocking()) {
+            // Test individual headers
+            HttpRequest request = client.get("/");
+            request.addHeader(TE, "foo");
+            request.addHeader(TE, TRAILERS);
+            request.addHeader(TE, "bar");
```

Comment:
In the next Netty version 4.1.84 the `DefaultHttp2Headers` impl will throw when users add `TE` value different from `trailers`. To keep the test coverage of `H2ToStH1Utils`, consider using `DefaultHttpHeadersFactory` when `h2PriorKnowledge == true` instead of `H2HeadersFactory`. 

Use-cases for `H2HeadersFactory` (backed by `DefaultHttp2Headers` from Netty) are already covered by tests in Netty.